### ISSUE TITLE
PCTLoadSchema regression, error on load of empty df.

### ISSUE THREAD

**Issue by cverbiest** [Riverside-Software/pct#124]
Target

```
<target name="load_food_incdf" depends="dump_food_incdf" description="Load incremental df">
  <PCTLoadSchema srcFile="syncfood/incr.df" dlcHome="${DLC}" cpstream="${cpstream}" >
    <DbConnection refid="LISAfood" />
  </PCTLoadSchema>
</target>
```

df

```
.
PSC
cpstream=utf-8
.
0000000002
```

**With PCT-187 I get a succesfull build**
load_food_incdf:
[PCTLoadSchema] Loading /usr2/cce/appl/antwork/syncfood/incr.df in database
**[PCTLoadSchema] 0 errors and 0 warnings during load**
[PCTLoadSchema] Log file can be found in /data/antwork/appl/LISAfood.e.16122015.33346.42

**With PCT-202 I get a failed build**
load_food_incdf:
[PCTLoadSchema] Loading /usr2/cce/appl/antwork/syncfood/incr.df in database
[PCTLoadSchema] Input file is empty or invalid.
**[PCTLoadSchema] 2 errors and 0 warnings during load**
[PCTLoadSchema] Log file can be found in /data/antwork/appl/LISAfood.e.16122015.33688.48

I think getting an error for an empty df can be useful but I'd prefer to see an option to have a choice between old and new PCT behaviour

**Comment by jakejustus** [comment]
Since PCT-192, when working with 11.3+, the new DF load procedures (from PSC) are being used. This procedure returns an error when the DF file is empty. Will have a look if it's possible to catch this kind of problem easily.

**Comment by djbrown92** [comment]
Hello again, Gilles. Now that we are using Progress OE11 I decided to revisit the issue I reported in https://github.com/Riverside-Software/pct/issues/117. The good news is that I now get a build failure when the schema is in use and there are schema changes that must be applied. 

However, I am now encountering this issue ("Input file is empty or invalid."), on a subsequent build, when there are no schema changes to apply! This is causing an unwanted build failure.

My version details for this are:

 PCT Version : pct-207-master-daf7eeaeb8310f3ee4e863656b6dc95b8abccec7
 OpenEdge Release 11.6 as of Fri Oct 16 19:01:51 EDT 2015

**Comment by gquerret** [comment]
Fixed in master

**Comment by cverbiest** [comment]
I have an alternative approach for this one , bit late as the case is closed but I'll add anyway it for future reference
Uses https://ant.apache.org/manual/ifunless.html 
```
<project name="Load df" xmlns:if="ant:if" xmlns:unless="ant:unless" >
```
* create a file matching the content of an empty incremental df
* compare the files
* only load the df if the file is not empty

```
    <target name="load_incdf" depends="dump_incdf" description="Load incremental df">
        <echo file="sync/empty.df">.${line.separator}PSC${line.separator}cpstream=utf-8${line.separator}.${line.separator}0000000002${line.separator}</echo>
        <condition property="emptydf" value="true">
            <filesmatch file1="sync/incr.df" file2="sync/empty.df"/>
        </condition>
        <echo if:set="emptydf">Incrementral df is empty</echo>
        <PCTLoadSchema unless:set="emptydf" srcFile="sync/incr.df" dlcHome="${DLC}" cpstream="${cpstream}">
            <DbConnection refid="targetdb"/>
            <env key="TERM" value="ansi" />
        </PCTLoadSchema>
    </target>

```

**Comment by gquerret** [comment]
@cverbiest Did you try the fix in 911530d2558d16051d473148afd4b7c07ea0ddc1 ?
Or was there something missing / not working correctly ?
